datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1125,6 +1126,14 @@ impl LogicalPlan {
         }
     }
 
+    /// These are invariants to hold true for each logical plan.
+    pub fn assert_invariants(&self) -> Result<()> {
+        // Refer to <https://datafusion.apache.org/contributor-guide/specification/invariants.html#relation-name-tuples-in-logical-fields-and-logical-columns-are-unique>
+        assert_unique_field_names(self)?;
```

Comment:
I had forgotten about that documentation 🤔 

Maybe it would be a good thing to inline that information into the code rather than in a separate document that might be harder to find (not for you, I am just musing here for my own benefit)

Reply:
I've added this to the followup items list: https://github.com/apache/datafusion/issues/13652#issuecomment-2550443777